components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/RequestObjectValidatorUtil.java

```diff
@@ -159,6 +172,37 @@ public static boolean isSignatureVerified(SignedJWT signedJWT, String jwksUri) t
 
     }
 
+    /**
+     * Validate the signature algorithm according to FAPI specification.
+     * According to FAPI, signature algorithm should be PS256 or ES256.
+     * <a href="https://openid.net/specs/openid-financial-api-part-2-1_0.html#algorithm-considerations">...</a>
+     *
+     * @param clientId  client id
+     * @param algorithm signature algorithm
+     * @return is valid signature algorithm
+     * @throws RequestObjectException if an error occurred while getting the service provider
+     */
+    private static boolean isValidFAPISignatureAlgorithm(String clientId, String algorithm)
+            throws RequestObjectException {
+
+        String requestObjSignatureAlgorithms = getSpProperty(clientId, Constants.REQUEST_OBJECT_SIGNING_ALG);
```

Comment:
Do we have a requirement to store the request object signing alg per SP? or even at a server level. since this is a jwt created by the client why not simply accept the jwt if its one of the allowed algorithms?

Reply:
DCR specification mentions request_object_signing_alg. So it just ensure whether the actual signing algo is the one registered with.
For FAPI scenarios we have added a server level config as well to retrieve the FAPI compliant algorithms.
For general use cases we need to give precedence to the app level attribute or else validate against the server level

Reply:
Ack.
Since we are allowing to store the request obj sig alg we need to validate against that regardless of the FAPI conformance. Then for fapi conformance case check the req obj sig alg against allowed fapi algs.

Reply:
fixed with 734bb950389d77ac4eae85e698cfce87cfd83209